Mage.Sets/src/mage/cards/i/IndominusRexAlpha.java

```diff
@@ -0,0 +1,183 @@
+            if (!(counter instanceof AbilityCounter)) {
+                throw new IllegalArgumentException(counterType + " is not an ability counter");
+            }
+        }
+    }
+
+    protected String getFirstChoiceAsString(){
+        return copyableCounters.toArray(new CounterType[0])[0].getName();
+    }
+
+    protected String getOtherChoicesAsString(){
+        StringBuilder sb = new StringBuilder();
+        CounterType[] choicesArray = copyableCounters.toArray(new CounterType[0]);
+
+        // Skip first counter name - flying
+        for (int i = 1; i < copyableCounters.size(); i++){
+            String separator = ", ";
+            if (i == copyableCounters.size()-2){
+                separator = ", and ";
+            } else if (i == copyableCounters.size()-1){
+                separator = "";
+            }
+            sb.append(choicesArray[i].getName()).append(separator);
+        }
+        return sb.toString();
+    }
+
+    private IndominusRexAlphaCountersEffect(final IndominusRexAlphaCountersEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public IndominusRexAlphaCountersEffect copy() {
+        return new IndominusRexAlphaCountersEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+
+        // EntersBattlefieldAbility is static, see AddCountersSourceEffect
+        Permanent permanent = game.getPermanentEntering(source.getSourceId());
+        if (permanent == null) {
+            return true;
+        }
+
+        TargetCard target = new TargetDiscard(0, Integer.MAX_VALUE, StaticFilters.FILTER_CARD_CREATURE, controller.getId());
+        controller.choose(outcome, controller.getHand(), target, source, game);
+        List<UUID> chosenTargets = target.getTargets();
+
+        // the basic event is the EntersBattlefieldEvent, so use already applied replacement effects from that event
+        List<UUID> appliedEffects = (ArrayList<UUID>) this.getValue("appliedEffects");
+
+        boolean addedCounters = false;
+
+        for (CounterType counterType : copyableCounters) {
+
+            Counter counter = counterType.createInstance();
+            Ability abilityToCopy = ((AbilityCounter)(counter)).getAbility();
+            Class<? extends Ability> abilityClass = abilityToCopy.getClass();
+
+            boolean addedCounterType = false;
+
+            for (UUID targetId : chosenTargets) {
+
+                Card card = game.getCard(targetId);
+                if (card == null){
+                    continue;
+                }
+
+                for (Ability ability : card.getAbilities()) {
+                    if (abilityClass.isInstance(ability)){
+                        permanent.addCounters(counter, source.getControllerId(), source, game, appliedEffects);
+                        addedCounterType = true;
```

Comment:
No need in addedCounterType at all. Just set addedCounters field here.

Reply:
If i do that, then after the first counter type gets added, it will not know the difference between having added a counter of the current iterated type or having added counters at all--after finding the ability its looking for on one of the cards, it needs to break out of the ability loop as well as the discarded card loop. 

If it just sets the larger `addedCounters` to true, either it will think it found the counter type after looping over the abilities of just one card (assuming i keep the conditional break statement on line 175) resulting in it missing an ability counter even if a discarded card had that ability, or if i remove the outer break statement, it will add multiple counters of the same ability if multiple discarded cards have that ability.

Reply:
As i address another change, this might not matter lol

Reply:
Then all fine